src/lib/offline-api/index.ts

```diff
@@ -351,6 +413,33 @@ class OfflineAPI {
     return entries
   }
 
+  async getLinks (childId: string, locales: string[]): Promise<Link[]> {
```

Comment:
This is a super expensive operation.
Maybe we should index all links once when we initialize all the entries, then again it would make it more difficult to deal with updates to entries during the migration, which would also have to be added to the index.
I feel like for now this may be good enough, but I can already see a few people with larger spaces having performance problems due to repeatedly iterating every single entry in their entire space to look for links.

Reply:
Fully agree. We talked about this but weren't sure what the best method would be to do this. Especially considering that we do not want to penalize users that do not use this type of transform and that we have to make sure to keep this up to date like you said.

Also we thought that the PR is big enough as it is, so it might be better to have this as a separate improvement.

That being said we actually tested the migration on a big space with several thousand items and did not have a problem performance wise so far. Most of the time spent was actually at the beginning since we need to fetch all entries in the space to make this kind of calculation in the first place, which can take some time on a slow network.